repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/core/wrappers/response/GlassfishResponseWrapperTest.groovy

```diff
@@ -0,0 +1,100 @@
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package features.core.wrappers.response
+
+import org.openrepose.framework.test.ReposeConfigurationProvider
+import org.openrepose.framework.test.ReposeContainerLauncher
+import org.openrepose.framework.test.TestProperties
+import org.rackspace.deproxy.Deproxy
+import spock.lang.Shared
+import spock.lang.Specification
+
+class GlassfishResponseWrapperTest extends Specification {
+
+    static final int ERROR_CODE = 419
+    static final String ERROR_HEADER_NAME = "Error"
+    static final String ERROR_MESSAGE = "OHJEEZ"
+    static final String ERROR_BODY = "Your request is bad and you should feel bad!"
+
+    @Shared
+    Deproxy deproxy
+
+    @Shared
+    ReposeContainerLauncher glassfishLauncher
+
+    @Shared
+    TestProperties properties = new TestProperties(this.getClass().canonicalName.replace('.', '/'))
+
+    def setupSpec() {
+        def configurationProvider = new ReposeConfigurationProvider(properties)
+
+        def params = properties.defaultTemplateParams
+        params += [
+            clusterId   : 'cluster1',
+            nodeId      : 'node1',
+            errorCode   : ERROR_CODE,
+            errorMessage: ERROR_MESSAGE,
+            errorBody   : ERROR_BODY,
+            headerName  : ERROR_HEADER_NAME
+        ]
+        configurationProvider.applyConfigs("common", params)
+        configurationProvider.applyConfigs("features/core/wrappers/response/war", params)
+
+        deproxy = new Deproxy()
+        deproxy.addEndpoint(properties.targetPort)
+
+        glassfishLauncher = new ReposeContainerLauncher(
+            configurationProvider,
+            properties.glassfishJar,
+            params.clusterId as String,
+            params.nodeId as String,
+            properties.reposeGlassfishRootWar,
+            properties.reposePort)
+        glassfishLauncher.start()
+        glassfishLauncher.waitForDesiredResponseCodeFromUrl(reposeEndpoint, 200)
+    }
+
+    def cleanupSpec() {
+        deproxy?.shutdown()
+        glassfishLauncher?.stop()
+    }
+
+    def "a filter writes a header and response body then calls sendError on the response"() {
+        when: "a request is made to Repose"
+        def messageChain = deproxy.makeRequest(reposeEndpoint + "/scripting")
+
+        then: "the response code should be the error code, not a 500"
+        messageChain.receivedResponse.code as Integer == ERROR_CODE
+
+        and: "the status line message should be the error message"
+        messageChain.receivedResponse.message == ERROR_MESSAGE
+
+        and: "the response body should be the container-server error page"
+        (messageChain.receivedResponse.body as String).length() != 0
+        messageChain.receivedResponse.body as String != ERROR_BODY
+
+        and: "the header should be written"
+        messageChain.receivedResponse.headers.findAll(ERROR_HEADER_NAME).size() == 1
+    }
+
+    def getReposeEndpoint() {
```

Comment:
why?

Reply:
This is weird and not used.

Reply:
Oh, it's totally used, it's just used via Groovy magic. Instead of calling `getRequestEndpoint`, this is called everywhere you see `reposeEndpoint` as a variable.

As for why, it was just to be consistent with the `ReposeValveTest` which does the same thing. Even though I am not extending `ReposeValveTest` (because it looks like container support started to get added to that, but was never fully added... and I'm not sure why it was started to begin with), I wanted the test to be familiar. I suspect this was originally added to `ReposeValveTest` as a minor convenience.

https://github.com/rackerlabs/repose/blob/master/repose-aggregator/tests/functional-test-framework/src/main/groovy/org/openrepose/framework/test/ReposeValveTest.groovy#L150-L173

Reply:
That all being said, I'm fine with inlining this method if desired.

Reply:
Oh yes. The Groovy magic. I'm down with the inlining as well.

Reply:
yeah, inlining makes things clearer here